homeassistant/components/integration/sensor.py

```diff
@@ -314,3 +384,35 @@ def native_value(self) -> Decimal | None:
     def native_unit_of_measurement(self) -> str | None:
         """Return the unit the value is expressed in."""
         return self._unit_of_measurement
+
+    @property
+    def extra_state_attributes(self) -> dict[str, str] | None:
```

Comment:
Why does it have to be in the attributes if you are already using `extra_restore_state_data` ?

Reply:
To make it available to templates while the sensor is unavailable.

Most of the current open issues associated with this integration are associated to the unavailability state of the sensor. By providing it as an attribute users can get the last valid state and use in their calculations.

Reply:
removed the last_valid_state

Reply:
Why do we need to add source ID?

Reply:
I can't remember the use case, it has been for too long

Should I remove and document as a breaking change ? Better to create a separate PR for that no ?

Reply:
my bad, I didn't realize this is existing.